DP3TApp/SharedUI/Style/UIColor+NS.swift

```diff
@@ -11,23 +11,60 @@
 import UIKit
 
 extension UIColor {
-    public static var ns_red = UIColor(ub_hexString: "#e20008")!
-    public static var ns_text = UIColor(ub_hexString: "#4a4969")!.withHighContrastColor(color: .black)
+    
+    public static var ns_red: UIColor {
+        return UIColor().setColorsForTheme(lightColor: UIColor(ub_hexString: "#e20008")!, darkColor: UIColor(ub_hexString: "#FF3B42")!)
+    }
+    
+    // MARK: - Text color
 
+    public static var ns_text: UIColor {
+        return UIColor().setColorsForTheme(lightColor: UIColor.defaultTextColor, darkColor: UIColor.white.withHighContrastColor(color: .black))
+    }
+    
+    public static var defaultTextColor = UIColor.ns_darkBlueBackground.withHighContrastColor(color: .black)
+    
     public static var ns_blue = UIColor(ub_hexString: "#5094bf")!.withHighContrastColor(color: UIColor(ub_hexString: "#2769a3")!)
+    public static var ns_lightBlue = UIColor(ub_hexString: "#00a7d4")!.withHighContrastColor(color: UIColor(ub_hexString: "#59738A")!)
     public static var ns_blueBackground = UIColor(ub_hexString: "#eff5f9")!
     public static var ns_green = UIColor(ub_hexString: "#009e89")!.withHighContrastColor(color: UIColor(ub_hexString: "#007363")!)
-    public static var ns_greenBackground = UIColor(ub_hexString: "#e5f8f6")!
     public static var ns_purple = UIColor(ub_hexString: "#8d6a9f")!.withHighContrastColor(color: UIColor(ub_hexString: "#6e3f86")!)
+    public static var ns_greenBackground: UIColor = UIColor(ub_hexString: "#e5f8f6")!
+    
+    public static var moduleGreenBackground: UIColor {
+        return UIColor().setColorsForTheme(lightColor: UIColor(ub_hexString: "#e5f8f6")!, darkColor: UIColor(ub_hexString: "#16161A")!)
+    }
+        
     public static var ns_purpleBackground = UIColor(ub_hexString: "#f3f0f5")!
-
     public static var ns_darkBlueBackground = UIColor(ub_hexString: "#4a4969")!
-
-    public static var ns_background = UIColor.white
-    public static var ns_backgroundSecondary = UIColor(ub_hexString: "#f7f7f7")!
-    public static var ns_backgroundTertiary = UIColor(ub_hexString: "#efefef")!
+    
+    static var ns_background: UIColor {
+        return UIColor().setColorsForTheme(lightColor: UIColor.white, darkColor: UIColor.black)
+    }
+    
+    static var moduleBackground: UIColor {
+        return UIColor().setColorsForTheme(lightColor: UIColor.white, darkColor: UIColor(ub_hexString: "#1E1E23")!)
+    }
+    
+    static var ns_backgroundSecondary: UIColor {
+        return UIColor().setColorsForTheme(lightColor: UIColor(ub_hexString: "#f7f7f7")!, darkColor: UIColor.black)
+    }
+    
+    static var ns_backgroundTertiary: UIColor {
+        return UIColor().setColorsForTheme(lightColor: UIColor(ub_hexString: "#efefef")!, darkColor: UIColor(ub_hexString: "#1E1E23")!)
+    }
+    
+    static var disabledButtonBackground: UIColor {
```

Comment:
To keep stuff consistent this needs to be renamed to ns_disabledButtonBackground